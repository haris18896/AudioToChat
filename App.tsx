import React from 'react';

import { SafeAreaView, StyleSheet } from 'react-native';
import { ChatScreen } from './src/components';

function App() {
  return (
    <SafeAreaView style={styles.container}>
      <ChatScreen />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F8FAFC',
  },
});

export default App;
