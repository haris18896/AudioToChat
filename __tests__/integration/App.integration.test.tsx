import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../App';

// Mock SafeAreaProvider to avoid native dependencies
jest.mock('react-native-safe-area-context', () => ({
  SafeAreaProvider: ({ children }: any) => children,
}));

// Mock the ChatScreen component to avoid complex dependencies
jest.mock('../../src/components/ChatScreen', () => {
  const mockReact = require('react');
  return function MockChatScreen() {
    return mockReact.createElement('View', { testID: 'chat-screen' });
  };
});

describe('App Integration Tests', () => {
  it('should render the complete application without crashing', () => {
    expect(() => {
      const tree = renderer.create(<App />);
      tree.unmount();
    }).not.toThrow();
  });

  it('should create a valid component tree', () => {
    const tree = renderer.create(<App />);
    const instance = tree.getInstance();
    expect(instance).toBeDefined();
    tree.unmount();
  });

  it('should match snapshot structure', () => {
    const tree = renderer.create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
    tree.unmount();
  });
});
