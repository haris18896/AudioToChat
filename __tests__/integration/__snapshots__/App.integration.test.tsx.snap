// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Integration Tests should match snapshot structure 1`] = `null`;
